### PR TITLE
feature/mx 1382 merged items endpoint

### PR DESCRIPTION
**Pull request by rababerladuseladim**
# Changes
- merged endpoint returns actual merged items. was: extracted items presented as merged items

### CODE REVIEW

File: mex/backend/graph/cypher/fetch_merged_items.cql [hunk 1]
```diff
@@ -0,0 +1,69 @@
+<# Fetch merged items including their nested objects and referenced identifiers.
```
Comment (cutoffthetop) [inline mex/backend/graph/cypher/fetch_merged_items.cql:1]: nicely done! this was quite complex i imagine, but looks good. i created a follow-up in MX-1693 so we can try to combat some of the repetition across all queries

File: mex/backend/rules/helpers.py [hunk 2]
```diff
@@ -23,22 +22,21 @@
...
 
-    if len(rules) != 3:
+    if len(raw_rules) != 3:
         raise MExError("inconsistent rule item count")
 
-    for rule in rules:
+    for rule in raw_rules:
         for field_name, model_class_lookup in MODEL_CLASS_LOOKUP_BY_FIELD_NAME.items():
-            if model := model_class_lookup.get(rule.get("entityType")):
+            if model := model_class_lookup.get(str(rule.get("entityType"))):
```
Comment (cutoffthetop) [inline mex/backend/rules/helpers.py:39]: when was it not a string?

File: mex/backend/graph/cypher/fetch_merged_items.cql [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+Args:
+    filter_by_query_string: Whether the final query should accept a full text query string
+    filter_by_stable_target_id: Whether the final query should filter by stableTargetId
+
+Returns:
+    total: Count of all items found by this query
+    items: List of merged items, each item has the following attributes:
+        entityType: the type of the merged item
+        identifier: the identifier of the merged item
+        components: The rule and / or extracted items with the stableTargetId of this
```
Comment (cutoffthetop) [inline mex/backend/graph/cypher/fetch_merged_items.cql:18]: i thought we called those _fragments_ in some repo or documentation. but i couldn't find any search hits, so maybe it was just in my head 😄

File: mex/backend/merged/helpers.py [hunk 4]
```diff
@@ -125,33 +141,49 @@ def search_merged_items_in_graph(
...
+            component
+            for component in item["components"]
+            if component["entityType"] in RULE_MODEL_CLASSES_BY_NAME
+        ]
+        if len(rules_raw) == 3:
+            rule_set_response = transform_raw_rules_to_rule_set_response(rules_raw)
+        elif len(rules_raw) == 0:
+            rule_set_response = None
+        else:
+            raise MExError(
```
Comment (cutoffthetop) [inline mex/backend/merged/helpers.py:175]: some test coverage here please

**Review by cutoffthetop — CHANGES_REQUESTED** [review]
very nice 🎇 

just a small test and a main-merge missing

File: mex/backend/rules/helpers.py (hunk 2 above)
Comment (rababerladuseladim) [reply mex/backend/rules/helpers.py:39]: This is merely to silence mypy, due to the type of rule being `dict[str, Any]`.

File: mex/backend/graph/cypher/fetch_merged_items.cql (hunk 1 above)
Comment (rababerladuseladim) [reply mex/backend/graph/cypher/fetch_merged_items.cql:1]: Thank you 😊 I certainly learned some cypher along the way.

File: mex/backend/merged/helpers.py [hunk 5]
```diff
@@ -125,33 +141,49 @@ def search_merged_items_in_graph(
...
         try:
-            items.append(merged_model_adapter.validate_python(item))
+            items.append(
+                create_merged_item(
+                    identifier=item["identifier"],
+                    extracted_items=extracted_items,
+                    rule_set=rule_set_response,
+                )
+            )
         except ValidationError as error:
```
[the remark this answers is not in the document]
Comment (rababerladuseladim) [reply mex/backend/merged/helpers.py]: I removed the exception catching in c6abeb6e332b88490e7b2475e12d5c8dca944065

File: mex/backend/merged/helpers.py (hunk 4 above)
Comment (rababerladuseladim) [reply mex/backend/merged/helpers.py:174]: done in d3208335baa4c8e4e9c38825c8e9941dbb06f464

**Review by cutoffthetop — APPROVED** [review]

**Pull request merged**